roles/ovirt/tasks/main.yml

```diff
@@ -0,0 +1,101 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.1
+
+- name: Check if variables are defined and have specific values
+  ansible.builtin.assert:
+    fail_msg: >-
+      One or more variables are not defined or do not have expected values!
+    that:
+      - (cifmw_ovirt_engine.url is defined) or (lookup('env','OVIRT_URL'))
+      - (cifmw_ovirt_engine.username is defined) or (lookup('env','OVIRT_USERNAME'))
+      - (cifmw_ovirt_engine.password is defined) or (lookup('env','OVIRT_PASSWORD'))
+      - (cifmw_ovirt_engine.insecure is defined) or (lookup('env','OVIRT_INSECURE'))
+      - (cifmw_ovirt_cluster_name is defined) or (lookup('env','OVIRT_CLUSTER_NAME'))
+      - cifmw_ovirt_template_rhel_guest_image is defined
+
+- name: Ensure artifacts directory present
+  ansible.builtin.file:
+    path: "{{ cifmw_ovirt_artifacts_basedir }}"
+    state: directory
+
+- name: Check if the ssh keys already exist
+  ansible.builtin.stat:
+    path: "{{ cifmw_ovirt_artifacts_basedir }}/id_ecdsa"
+  register: check_id_ecdsa
+
+- name: Create the ssh keys
+  when: not check_id_ecdsa.stat.exists
+  community.crypto.openssh_keypair:
+    path: "{{ ansible_user_dir  }}/.ssh/id_{{ cifmw_ovirt_ssh_key_type }}"
+    type: "{{ cifmw_ovirt_ssh_key_type }}"
+    size: "{{ cifmw_ovirt_ssh_key_size }}"
+
+- name: Get local public key
+  register: _ssh_key_slurp
+  ansible.builtin.slurp:
+    src: "{{ ansible_user_dir  }}/.ssh/id_{{ cifmw_ovirt_ssh_key_type }}.pub"
```

Comment:
If you only need the public key here the slurp is unnecessary, as `community.crypto.openssh_keypair` returns it directly.

Reply:
Removed.